sympy/geometry/point.py

```diff
@@ -80,49 +82,225 @@ class Point(GeometryEntity):
         return item in self.args
 
-    def is_concyclic(*args):
-        # Coincident points are irrelevant and can confuse this algorithm.
-        # Use only unique points.
-        args = list(set(args))
-        if not all(isinstance(p, Point) for p in args):
-            raise TypeError('Must pass only Point objects')
+    @staticmethod
+    def are_coplanar(*points):
+        """Return True if the points lie in a unique plane.
+
+        All two dimensional points are coplanar: they lie in the
+        x-y plane.  Three dimensional points are coplanar only
+        if they are not collinear and thus define a unique plane.
+
+        Parameters
+        ==========
+
+        A set of points
+
+        Raises
+        ======
+
+        ValueError : if there is a dimension mismatch in point given.
+        NotImplementedError : for dimensions higher than 3
+
+        Returns
+        =======
+
+        boolean
+
+        Examples
+        ========
 
+        >>> from sympy import Point3D
+        >>> p1 = Point3D(1, 2, 2)
+        >>> p2 = Point3D(2, 7, 2)
+        >>> p3 = Point3D(0, 0, 2)
+        >>> p4 = Point3D(1, 1, 2)
+        >>> Point3D.are_coplanar(p1, p2, p3, p4)
+        True
+        >>> p5 = Point3D(0, 1, 3)
+        >>> Point3D.are_coplanar(p1, p2, p3, p5)
+        False
+
+        """
+        if not points:
+            return False
+        points = [Point(i) for i in points]
+        dim = points[0].ambient_dimension
+        if not all(i.ambient_dimension == dim for i in points):
+            raise ValueError('dimensions of all Points must be the same')
+        if points[0].ambient_dimension < 4:
+            cls = points[0]
+            return cls.are_coplanar(*points)
+        raise NotImplementedError('dimension higher than 3')
+
+    def is_concyclic(self, *points):
+        """Do `self` and the given sequence of points lie in a circle?
+
+        Returns True if the set of points are concyclic and
+        False otherwise.
+
+        Parameters
+        ==========
+
+        points : sequence of Point
+
+        Raises
+        ======
+
+        NotImplementedError : raised when the points are planar but have
+            a dimension other than 2
+
+        Returns
+        =======
+
+        is_concyclic : boolean
+
+
+        Examples
+        ========
+
+        >>> from sympy import Point
+        >>> from sympy.abc import x
+        >>> p1, p2 = Point(0, 0), Point(1, 1)
+        >>> p3, p4, p5 = Point(2, 2), Point(x, x), Point(1, 2)
+        >>> Point.is_collinear(p1, p2, p3, p4)
+        True
+        >>> Point.is_collinear(p1, p2, p3, p5)
+        False
+
+        """
+        args = (self,) + points
+        args = Point.normalize_dimensions(*[Point(p) for p in args])
+        args = list(uniq(args))
+        if not Point.are_coplanar(*args):
+            return False
+        if args[0].ambient_dimension != 2:
+            raise NotImplementedError
         return args[0].is_concyclic(*args[1:])
```

Comment:
In this case, `Point.is_concyclic` should raise a `NotImplimentedError` and `Point2D.is_concyclic` should overload that function.  In any case, we might as well make this fully general by adding a `is_cospherical` that works in n dimensions and then `is_concyclic` would be `is_cospherical & is_coplanar`.